### ISSUE TITLE
JaxRS-Client failed to create in Executor-Thread

### ISSUE THREAD

**Issue by dominikrebhan** [OpenLiberty/open-liberty#14398]
**Describe the bug**
When trying to create a ClientBuilder inside a managed thread (ManagedExecutorService) an exception is raised. This worked until 20.0.0.7 without any problems. 

[INFO] [err] java.lang.RuntimeException: java.lang.ClassNotFoundException: org.glassfish.jersey.client.JerseyClientBuilder cannot be found by com.ibm.websphere.javaee.jaxrs.2.1_1.0.45.cl201020200915-1100
[INFO] [err] 	at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:110)
[INFO] [err] 	at de.liberty.test.Startup.lambda$init$0(Startup.java:19)
[INFO] [err] 	at com.ibm.ws.threading.internal.PolicyTaskFutureImpl.run(PolicyTaskFutureImpl.java:715)
[INFO] [err] 	at [internal classes]
[INFO] [err] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[INFO] [err] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[INFO] [err] 	at java.lang.Thread.run(Thread.java:748)
[INFO] [err] Caused by: java.lang.ClassNotFoundException: org.glassfish.jersey.client.JerseyClientBuilder cannot be found by com.ibm.websphere.javaee.jaxrs.2.1_1.0.45.cl201020200915-1100
[INFO] [err] 	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:516)
[INFO] [err] 	at [internal classes]
[INFO] [err] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
[INFO] [err] 	at java.lang.Class.forName0(Native Method)
[INFO] [err] 	at java.lang.Class.forName(Class.java:264)
[INFO] [err] 	at javax.ws.rs.client.FactoryFinder.newInstance(FactoryFinder.java:111)
[INFO] [err] 	at javax.ws.rs.client.FactoryFinder.find(FactoryFinder.java:209)
[INFO] [err] 	at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:94)
[INFO] [err] 	... 7 more

**Steps to Reproduce**
Create an empty application with the following server.xml:

```xml
<server description="new server">
    <featureManager>
        <feature>cdi-2.0</feature>
        <feature>jaxrs-2.1</feature>
        <feature>concurrent-1.0</feature>
    </featureManager>
    <httpEndpoint id="defaultHttpEndpoint" httpPort="9080"/>
    <applicationManager autoExpand="true" />
    <managedExecutorService jndiName="concurrent/executor">
        <contextService>
            <jeeMetadataContext/>
        </contextService>
        <concurrencyPolicy max="5"/>
    </managedExecutorService>
</server>
```

Add the necessary beans.xml and the following class:

```java
@ApplicationScoped
public class Startup {

    @Resource(name = "concurrent/executor")
    ManagedExecutorService executorService;

    public void init(@Observes @Initialized(ApplicationScoped.class) Object init) {
        executorService.execute(() -> {
            try {
                System.out.print("--> " + ClientBuilder.newBuilder());
            }
            catch (Exception ex) {
                ex.printStackTrace();
            }
        });
    }
}
```

**Expected behavior**
The expected behavior is that the client can be successfully obtained and used within this scope.

**Diagnostic information:**
 - OpenLiberty Version: 20.0.0.10
 - Java Version: 
java version "1.8.0_231"
Java(TM) SE Runtime Environment (build 1.8.0_231-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.231-b11, mixed mode)
 - The messages.log from the test server
[messages.log](https://github.com/OpenLiberty/open-liberty/files/5350615/messages.log)

**Additional context**
I think it's a classloading issue with the META-INF/service folder. It may be linked with this commit: https://github.com/OpenLiberty/open-liberty/commit/da4e636ae51fa8b9b7e61d25485a22c73fc4c33f

When manually adding the META-INF/services back to the jaxrs.2.1_1.0.45.cl201020200915-1100 it works, also when setting the property in jvm.options: -Djavax.ws.rs.client.ClientBuilder=com.ibm.ws.jaxrs20.client.JAXRSClientBuilderImpl

**Comment by andymc12** [comment]
Hello,

I was able to reproduce the ClassNotFoundException using 20.0.0.8 and 20.0.0.10 as you mentioned.  The moving of the META-INF/services files is required in order to support products/applications that want to use the JAX-RS APIs from Open Liberty, but provide their own implementation.  That said, it was only due to a bug in 20.0.0.7 that you would have been able to load the ClientBuilder at all in this scenario.

The JAX-RS API depends primarily on the thread's context classloader to load the implementation class of the ClientBuilder interface. The thread context classloader is not propagated to the ManagedExecutorService because it is missing the `<classLoaderContext/>` element in the config.  Once that is applied, I was able to load the ClientBuilder in all versions of Liberty that I tried with (7, 8, and 10).  That setting will ensure that the thread context is propagated to the executor service so that those threads are able to load Liberty's implementation of the ClientBuilder (and other Java EE APIs).

Here is the server.xml that I used to make this work:
```
<server description="new server">
    <featureManager>
        <feature>cdi-2.0</feature>
        <feature>jaxrs-2.1</feature>
        <feature>concurrent-1.0</feature>
    </featureManager>
    <httpEndpoint id="defaultHttpEndpoint" httpPort="9080"/>
    <applicationManager autoExpand="true" />
    <managedExecutorService jndiName="concurrent/executor">
        <contextService>
            <jeeMetadataContext/>
            <classloaderContext/>
        </contextService>
        <concurrencyPolicy max="5"/>
    </managedExecutorService>
</server>
```

Hope this helps!
